partner_auto_archive/models/res_partner.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2024 ForgeFlow S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class Partner(models.Model):
+    _inherit = "res.partner"
+
+    auto_archive = fields.Boolean(
+        help="Archive contact automatically after a period of time"
+    )
+
+    @api.model
+    def _auto_archive_contacts(self):
+        contacts_to_archive = self.env["res.partner"].search(
+            [("auto_archive", "=", True)]
+        )
+        contacts_to_archive.action_archive()
```

Comment:
Perhaps a good improvement could be unsetting the `auto_archive` after running the `action_archive()` just so that if the partner gets unarchived in the future, the user doesn't have manually unmark the `auto_archive`.
Just as an nice to have, not needed for the funcionality of the module.

Reply:
I think it is a good idea. I will add it. Thanks!